Makefile

```diff
@@ -102,11 +118,28 @@ ifneq ("$(wildcard $(HAL_LIB_DIR)/*.so)","")
 	cp $(HAL_LIB_DIR)/*.so* $(BIN_DIR)
 endif
 
+test_tmp_fyaml: $(OBJS_TEST_YAML_TMP)
+	@echo -e ${GREEN}TARGET_EXEC [$(TARGET_TMP_FYAML_EXEC)]${NC}
+	@echo -e ${GREEN}Linking $@ $(BUILD_DIR)/$(TARGET_TMP_FYAML_EXEC)${NC}
+	@$(CC) $(OBJS_TEST_YAML_TMP) -o $(BUILD_DIR)/$(TARGET_TMP_FYAML_EXEC) $(XLDFLAGS) $(KCFLAGS) $(XCFLAGS)
+	@$(MKDIR_P) $(BIN_DIR)
+	@cp $(BUILD_DIR)/$(TARGET_TMP_FYAML_EXEC) $(BIN_DIR)
+
+ut_test_kvp : $(OBJS_TEST_YAML)
```

Comment:
u shouldn't need to-do this, you just add a new test into `tests/main.c`

when you run `tests/Makefile` since it looks for *.c anytning under tests/*c should be caught.